plugins/autoscroll.koplugin/main.lua

```diff
@@ -0,0 +1,161 @@
+    UIManager.event_hook:registerWidget("InputEvent", self)
+    self.autoscroll_sec = self.settings
+    self.ui.menu:registerToMainMenu(self)
+    self:_deprecateLastTask()
+    self:_start()
+end
+
+function AutoScroll:onInputEvent()
+    logger.dbg("AutoScroll: onInputEvent")
+    self.last_action_sec = os.time()
+end
+
+-- We do not want auto scroll procedure to waste battery during suspend. So let's unschedule it
+-- when suspending and restart it after resume.
+function AutoScroll:onSuspend()
+    logger.dbg("AutoScroll: onSuspend")
+    self:_deprecateLastTask()
+end
+
+function AutoScroll:onResume()
+    logger.dbg("AutoScroll: onResume")
+    self:_start()
+end
+
+function AutoScroll:addToMainMenu(menu_items)
+    menu_items.autoscroll = {
+        text_func = function() return self:_enabled() and T(_("Autoscroll (%1 s)"), self.autoscroll_sec)
+            or _("Autoscroll") end,
+        checked_func = function() return self:_enabled() end,
+        callback = function(menu)
+            local Screen = Device.screen
+            local SpinWidget = require("ui/widget/spinwidget")
+            local curr_items = G_reader_settings:readSetting("autoscroll_timeout_seconds") or 30
+            local autoscroll_spin = SpinWidget:new {
+                width = Screen:getWidth() * 0.6,
+                value = curr_items,
+                value_min = 0,
+                value_max = 240,
+                value_hold_step = 5,
+                ok_text = _("Set timeout"),
+                cancel_text = _("Disable"),
+                title_text = _("Timeout in seconds"),
+                cancel_callback = function()
+                    self.enabled = false
+                    G_reader_settings:flipFalse("autoscroll_enabled")
+                    self:_deprecateLastTask()
+                    menu:updateItems()
+                end,
+                callback = function(autoscroll_spin)
+                    self.autoscroll_sec = autoscroll_spin.value
+                    G_reader_settings:saveSetting("autoscroll_timeout_seconds", autoscroll_spin.value)
+                    self.enabled = true
+                    G_reader_settings:flipTrue("autoscroll_enabled")
+                    self:_deprecateLastTask()
+                    self:_start()
+                    menu:updateItems()
+                end,
+            }
+            UIManager:show(autoscroll_spin)
+        end,
+        hold_callback = function(menu)
+            local Screen = Device.screen
+            local SpinWidget = require("ui/widget/spinwidget")
+            local curr_items = G_reader_settings:readSetting("autoscroll_timeout_seconds") or 30
+            local autoscroll_spin = SpinWidget:new {
+                width = Screen:getWidth() * 0.6,
+                value = curr_items,
+                value_min = -20,
+                value_max = 20,
+                precision = "%.2f",
+                value_step = .1,
+                value_hold_step = .5,
```

Comment:
So, 1 = 1 page, 0.3 = 30% of the page?
Do value_min/max = 20, or anything > 1, has some use case?

Reply:
I think so, yes. Or if not it's close enough for it not to matter. One would just decrease or increase the value to achieve the desired effect.

I just noticed something odd though. In paged mode fractional values result in an instant end of book dialog. And related, sending this event when one's already showing pops up more on top of it.

Reply:
What happens when you're doing a dict lookup and enjoying the result? The page still scroll in the background without any issue?

Might need to pause this autoscroll whenever the readerview is not the top one ? :|

Reply:
> What happens when you're doing a dict lookup and enjoying the result? The page still scroll in the background without any issue?

If you're doing it really slowly without interacting, sure.

> Might need to pause this autoscroll whenever the readerview is not the top one ? :|

Is that currently possible?

Reply:
Dunno :) probably not.
Might need to ask UIManager if there's something in the stack above the reader...
Or have UIManager send some event when covered/uncovered, when a widget has requested it (and give the names of the events, or some fixed event with the widget name/id :)
Probably quite some work, for a little plugin :) (pinging @NiLuJe if he sees something easy.)

Reply:
The least intrusive would probably to add something to the effect of "am I on top of the stack", and query that from the plugin as-needed?

Reply:
(That might have to be sneakier: am I the highest *visible* widget on the stack).

Or not, Been a while since I played w/ UIManager.

Otherwise, possibly as easy as checking if `self._window_stack[1].widget == widget` ;).

Reply:
The top of the stack is probably queried easily enough, but ReaderUI isn't the top of the stack. There's statusbars, bookmarks, and all kinds of stuff like that on top of it.

Reply:
Or just a simple method UIManager:getTopWidgetId() => "reader" or "#1234ab32" that the widget could just call before deciding to send the onGotoViewRel.